app/src/main/java/com/example/woloxapp/ui/login/LoginViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package com.example.woloxapp.ui.login
+
+import android.app.Application
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.woloxapp.R
+
+class LoginViewModel(private val applicationCustom: Application) : AndroidViewModel(applicationCustom) {
+    companion object {
+        const val USERNAME = R.string.username.toString()
+        const val PASSWORD: String = R.string.password.toString()
+        const val SHARED_PREFERENCES_USERNAME: String = R.string.shared_preferences_username.toString()
```

Comment:
be careful with this, since accessing to resources retrieves an id representing such string. After this, you should obtain it using the context

Reply:
To clarify, R.string.username.toString() is something like "948172". If you want to have a reference to sharedPreferences keys, its not necessary to use the string resources path (moreover, its not recommended). The string resources path should be used only for UI related stuff. In this case, you may simply declare the string you prefer